lib/circuit.js

```diff
@@ -640,7 +640,14 @@ class CircuitBreaker extends EventEmitter {
 
 function handleError (error, circuit, timeout, args, latency, resolve, reject) {
   clearTimeout(timeout);
+
+  if (circuit.options.errorFilter(error)) {
+    circuit.emit('success', error, latency);
+    return reject(error);
```

Comment:
if you've filtered the error,  and emitting success,  why would you reject the promise.  i thought this would have resolved?

Reply:
That is a good question. It's complicated.

In the current version, if the error is filtered, but there is no fallback function, no `failure` event is emitted, but a rejected `Promise` is returned. If there is a fallback function, a resolved `Promise` is always returned. In this case, the behavior has been changed so that the fallback function is never called in the event of a filtered error. To remain partially consistent with past behavior, then the promise should be rejected since no fallback is applied.

Reply:
it just seems weird to emit success but reject

Reply:
Agree... what do you think the correct approach is? Always resolve when an error is filtered? I'm OK with that, but it will be at least a minor version bump.

Reply:
@lance Do you mean resolve when there's a fallback?

Reply:
I would vote for, if a error happens, but there is an errorFilter setup, then we should be resolving

If it has to be a major bump, thats ok.  Is error filter a fairly new addition.  i can't remember when it was added

Reply:
@lholmquist But resolve with what result? If there's a fallback we resolve with the fallback result, but if there isn't, then there is no result to resolve with, in this case, I think we should reject with the error, so the application can handle it.

Reply:
Yes, i think that sounds correct

Reply:
I think I'm good with this (until I'm not - lol).